docs/tutorial/native-code-and-electron.md

```diff
@@ -0,0 +1,375 @@
+// This will look for the compiled .node file in various places
+const bindings = require('bindings')
+const native = bindings('my_addon')
+
+// Create a nice JavaScript wrapper
+class MyNativeAddon extends EventEmitter {
+  constructor () {
+    super()
+
+    // Create an instance of our C++ class
+    this.addon = new native.MyAddon()
+  }
+
+  // Wrap the C++ method with a nicer JavaScript API
+  helloWorld (input = '') {
+    if (typeof input !== 'string') {
+      throw new TypeError('Input must be a string')
+    }
+    return this.addon.helloWorld(input)
+  }
+}
+
+// Export a singleton instance
+if (process.platform === 'win32' || process.platform === 'darwin' || process.platform === 'linux') {
+  module.exports = new MyNativeAddon()
+} else {
+  // Provide a fallback for unsupported platforms
+  console.warn('Native addon not supported on this platform')
+
+  module.exports = {
+    helloWorld: (input) => `Hello from JS! You said: ${input}`
+  }
+}
+```
+
+This JavaScript wrapper:
+
+1. Uses `bindings` to load our compiled native addon
+2. Creates a class that extends EventEmitter (useful for future extensions that might emit events)
+3. Instantiates our C++ class and provides a simpler API
+4. Adds some input validation on the JavaScript side
+5. Exports a singleton instance of our wrapper
+6. Handles unsupported platforms gracefully
+
+### Building and Testing the Addon
+
+Now we can build our native addon:
+
+```sh
+npm run build
+```
+
+This will run `node-gyp configure` and `node-gyp build` to compile our C++ code into a `.node` file.
+Let's create a simple test script to verify everything works. Create `test.js` in the project root:
+
+```sh
+// Load our addon
+const myAddon = require('./js')
+
+// Try the helloWorld function
+const result = myAddon.helloWorld('This is a test')
+
+// Should print: "Hello from C++! You said: This is a test"
+console.log(result)
+```
+
+Run the test:
+
+```sh
+node test.js
+```
+
+If everything works correctly, you should see:
+
+```txt
+Hello from C++! You said: This is a test
+```
+
+### Using the Addon in Electron
+
+To use this addon in an Electron application, you would:
+
+1. Include it as a dependency in your Electron project
+2. Build it targeting your specific Electron version. `electron-forge` handles this step automatically for you - for more details, see [Native Node Modules](./using-native-node-modules.md).
+3. Import and use it just like any other module in a process that has Node.js enabled.
+
+```js @ts-expect-error=[2]
+// In your main process
+const myAddon = require('my-native-addon')
+console.log(myAddon.helloWorld('Electron'))
+```
+
+## 4) References and Further Learning
+
+Native addon development can be written in several languages beyond C++. Rust can be used with crates like [`neon`](https://neon-rs.dev/) or [`node-bindgen`](https://github.com/infinyon/node-bindgen). Objective-C/Swift can be used through Objective-C++ on macOS.
```

Comment:
I think https://github.com/napi-rs/napi-rs is also a popular one, not sure if we should mention it as well.

Reply:
Sounds good!